component-api/src/main/java/org/talend/sdk/component/api/record/Schema.java

```diff
@@ -453,6 +543,101 @@ static String sanitizeConnectionName(final String name) {
+         * @param fields the fields ordering
+         * @return the order EntriesOrder
+         */
+        public static EntriesOrder of(final String fields) {
+            return new EntriesOrder(fields);
+        }
+
+        public EntriesOrder(final String fields) {
+            if (fields == null) {
+                fieldsOrder = emptyList();
+            } else {
+                fieldsOrder = Arrays.stream(fields.split(",")).collect(Collectors.toList());
+            }
+        }
+
+        /**
+         * Move a field after another one.
+         *
+         * @param after the field name reference
+         * @param name the field name
+         * @return this EntriesOrder
+         */
+        public EntriesOrder moveAfter(final String after, final String name) {
+            if (getFieldsOrder().indexOf(after) == -1) {
+                throw new IllegalArgumentException(String.format("%s not in schema", after));
+            }
+            getFieldsOrder().remove(name);
+            int destination = getFieldsOrder().indexOf(after);
+            if (!(destination + 1 == getFieldsOrder().size())) {
+                destination += 1;
+            }
+            getFieldsOrder().add(destination, name);
+            return this;
+        }
+
+        /**
+         * Move a field before another one.
+         *
+         * @param before the field name reference
+         * @param name the field name
+         * @return this EntriesOrder
+         */
+        public EntriesOrder moveBefore(final String before, final String name) {
+            if (getFieldsOrder().indexOf(before) == -1) {
+                throw new IllegalArgumentException(String.format("%s not in schema", before));
+            }
+            getFieldsOrder().remove(name);
+            getFieldsOrder().add(getFieldsOrder().indexOf(before), name);
+            return this;
+        }
+
+        /**
+         * Swap two fields.
+         *
+         * @param name the field name
+         * @param with the other field
+         * @return this EntriesOrder
+         */
+        public EntriesOrder swap(final String name, final String with) {
+            Collections.swap(getFieldsOrder(), getFieldsOrder().indexOf(name), getFieldsOrder().indexOf(with));
+            return this;
+        }
+
+        public String toFields() {
+            return getFieldsOrder().stream().collect(Collectors.joining(","));
+        }
+
+        @Override
+        public int compare(final Entry e1, final Entry e2) {
+            final String name1 = e1.getName();
+            final String name2 = e2.getName();
+            if (getFieldsOrder().contains(name1) && getFieldsOrder().contains(name2)) {
+                return getFieldsOrder().indexOf(name1) - getFieldsOrder().indexOf(name2);
+            }
+            if (getFieldsOrder().contains(name1)) {
+                return -1;
+            }
+            if (getFieldsOrder().contains(name2)) {
+                return 1;
+            }
```

Comment:
As on arryaList, indexof is coded as 
```java
public boolean contains(Object o) {
        return indexOf(o) >= 0;
}
```
it would be more efficient to calculate index before : 
```java
final int index1 =  getFieldsOrder().indexOf(name1);
final int index2 =  getFieldsOrder().indexOf(name2);
if (index1 >= 0 && index2 >= 0) {
 ...
```